ddtrace/contrib/jinja2/patch.py

```diff
@@ -0,0 +1,96 @@
+import jinja2
+from wrapt import wrap_function_wrapper as _w
+
+from ...ext import http
+from ...pin import Pin
+from ...utils.wrappers import unwrap as _u
+
+DEFAULT_TEMPLATE_NAME = "<template>"
+
+
+def patch():
+    if getattr(jinja2, "__datadog_patch", False):
+        # already patched
+        return
+    setattr(jinja2, "__datadog_patch", True)
+    Pin(app='jinja2', service=None, app_type=http.TEMPLATE).onto(jinja2)
```

Comment:
Might need input from someone else, but we probably don't want to do `service=None` here, and `app_type=http.TEMPLATE` probably doesn't make the most sense. We want that for `span_type`.

Input from @Kyle-Verhoog maybe?

Reply:
I agree `http.TEMPLATE` is for `span_type`, but I didn't know what to use :)

I don't know if it's possible to choose to report `jinja2` as a service or not using Pin. If `service=None` is wrong, then I can create a real service name.

Reply:
Since jinja2 should be being used in addition to another integration the service will be inherited from the parent: https://github.com/DataDog/dd-trace-py/blob/87b9ff74509fab917c7341f34ffabbbf14b74578/ddtrace/tracer.py#L182-L184 so we don't have to worry about `service=None`. The agent will reject spans without a service.

Our config api is stable enough now that I would prefer you use it (over the `Pin`) for allowing customization of the service (if someone ever wanted to set their own) as well as other trace metadata. Following along with what the requests integration does would be good :+1:.